master/buildbot/worker/kubernetes.py

```diff
@@ -0,0 +1,368 @@
+# This file is part of Buildbot. Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+from future.utils import PY2
+from future.utils import string_types
+
+import os
+import socket
+import tempfile
+
+from twisted.internet import defer
+
+from buildbot.interfaces import LatentWorkerFailedToSubstantiate
+from buildbot.util.httpclientservice import HTTPClientService
+from buildbot.util.logger import Logger
+from buildbot.worker.docker import DockerBaseWorker
+
+try:
+    import yaml
+    HAS_YAML = True
+except ImportError:
+    HAS_YAML = False
+
+if PY2:
+    FileNotFoundError = IOError
+
+
+log = Logger()
+
+
+class KubeClientService(HTTPClientService):
+
+    default_kubeconfig = '~/.kube/config'
+
+    kube_dir = '/var/run/secrets/kubernetes.io/serviceaccount/'
+
+    kube_namespace_file = os.path.join(kube_dir, 'namespace')
+    kube_token_file = os.path.join(kube_dir, 'token')
+    kube_cert_file = os.path.join(kube_dir, 'ca.crt')
+
+    config_loader_list = ['kube_config_loader', 'incluster_config_loader']
+
+    def __init__(self,
```

Comment:
Actually, this class is only a config loader.

You can call it KubernetesConfigLoader(), and have a makeHttpClientService() method.

so that client can do:
```
self._kube_config = KubernetesConfigLoader(...)
self._kube = self._kube_config.makeHttpClientService()
self._kube.get("...".format(namespace=self._kube_config.namespace))
```
With this method, you can use the fake httpclient service easily.
You can also patch the configloader for tests to skip configuration details